cdap-ui/app/cdap/components/PipelineConfigurations/Store/ActionCreator.js

```diff
@@ -72,6 +126,28 @@ const getMacrosResolvedByPrefs = (resolvedPrefs = {}, macrosMap = {}) => {
   return resolvedMacros;
 };
 
+const updatePreferences = () => {
+  let {runtimeArgs} = PipelineConfigurationsStore.getState();
+  let appId = PipelineDetailStore.getState().name;
+  let prefObj = {};
+  runtimeArgs
+    .pairs
+    .filter(runTimeArg => runTimeArg.key.length)
+    .forEach(runTimeArg => {
+      prefObj[runTimeArg.key] = runTimeArg.value;
+    });
```

Comment:
You can use `convertKeyValuePairsObjToMap()` for `KeyValueStoreActions` instead of this block of code.